src/data_source/fetching.rs

```diff
@@ -929,13 +1004,15 @@ where
         let mut prev_height = 0;
 
         for i in 0.. {
+            let minimum_block_height =
+                self.storage.read().await.pruned_height.unwrap_or(0) as usize;
             // Get the block height; we will look for any missing blocks up to `block_height`.
             let block_height = { self.storage.read().await.height } as usize;
```

Comment:
We should use the same lock on storage to read the minimum block height and block height. E.g.
```rust
let (minimum_block_height, block_height) = {
  let storage = self.stoage.read().await;
  ...
};
```